### PR TITLE
Configuration expansion

### PR DESCRIPTION
**Pull request by PaulFarver**
- Config file validation with gopkg.in/go-playground/validator.v9
- Expand config to include options to set major/minor labels
- Expand config to include pre-release and full-release strategy types
- Use logrus for logging
- Package the github client in a releaser which takes care of release logic based on configuration

```release-note
Support release strategy types in .ship-it
```

### CODE REVIEW

File: cmd/serve.go [hunk 1]
```diff
@@ -28,32 +28,35 @@ var serveCmd = &cobra.Command{
...
+		lvl, err := logrus.ParseLevel(viper.GetString("server.loglevel"))
+		if err != nil {
+			logger.Fatal(err)
+		}
+		logger.SetLevel(lvl)
+
+		switch viper.GetString("server.logformat") {
 		default:
-			lvl = log.INFO
+			logger.Warnf("Could not understand log format '%s'. Defaulting to text", viper.GetString("server.logformat"))
```
Comment (naueramant) [inline cmd/serve.go]: Might be a little hard to understand that default fall to the next for a default. Consider moving default to the bottom and explicitly set lvl.

Or disregard this comment ;)

File: internal/rest/handler.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+		return c.String(http.StatusAccepted, "Handling push event")
+	case *github.ReleaseEvent:
+		r, err := h.initReleaser(c, event, event.GetRepo(), event.GetRelease().GetTagName(), entry)
+		if err != nil {
+			entry.WithError(err).Error("Could not initialize releaser")
+
+			return err
+		}
+		go r.HandleRelease(event)
```
Comment (naueramant) [inline internal/rest/handler.go]: Should a context be provided here?

File: internal/rest/handler.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+	switch event := event.(type) {
+	case *github.PushEvent:
+		r, err := h.initReleaser(c, event, event.GetRepo(), event.GetHead(), entry)
+		if err != nil {
+			entry.WithError(err).Error("Could not initialize releaser")
+
+			return err
+		}
+		go r.HandlePush(event)
```
Comment (naueramant) [inline internal/rest/handler.go]: Should a context be provided here?

File: cmd/serve.go (hunk 1 above)
Comment (PaulFarver) [reply cmd/serve.go]: Noted, and implemented

File: internal/rest/handler.go (hunk 2 above)
Comment (PaulFarver) [reply internal/rest/handler.go]: I've added contexts throughout the call tree in these functions. I haven't implemented any deadlines or timeouts, though. I'm not sure what timeout would be appropriate here, so I'm leaving this open. Could potentially be configurable in the .ship-it file.

**Review by naueramant — APPROVED** [review]

**Comment by PaulFarver** [comment]
related to #19

**Pull request merged**